Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -2255,9 +2265,40 @@ - (void)_startTextManipulationsWithConfiguration:(_WKTextManipulationConfigurati
     });
 }
 
-static WebCore::TextManipulationItemIdentifier coreTextManipulationItemIdentifierFromString(NSString *identifier)
-{
-    return makeObjectIdentifier<WebCore::TextManipulationItemIdentifierType>(identifier.longLongValue);
+struct ItemIdentifiers {
+    WebCore::FrameIdentifier frameID;
+    WebCore::TextManipulationItemIdentifier itemID;
+};
+
+static std::optional<ItemIdentifiers> coreTextManipulationItemIdentifierFromString(NSString *identifier)
+{
+    String identifierString { identifier };
+    unsigned index = 0;
+    std::optional<uint64_t> processID;
+    std::optional<uint64_t> frameID;
+    std::optional<uint64_t> itemID;
+    for (auto token : StringView(identifierString).split('-')) {
+        switch (index) {
+        case 0:
+            processID = parseInteger<uint64_t>(token);
+            break;
+        case 1:
+            frameID = parseInteger<uint64_t>(token);
+            break;
+        case 2:
+            itemID = parseInteger<uint64_t>(token);
+            break;
+        default:
+            return std::nullopt;
+        }
+        ++index;
+    }
+    if (!processID || !frameID || !itemID)
+        return std::nullopt;
+
+    return ItemIdentifiers { WebCore::ProcessQualified(makeObjectIdentifier<WebCore::FrameIdentifierType>(*frameID),
+        makeObjectIdentifier<WebCore::ProcessIdentifierType>(*processID)),
+        makeObjectIdentifier<WebCore::TextManipulationItemIdentifierType>(*itemID) };
```

Comment:
Do we need to validate the ObjectIdentifiers (not 0)?

Reply:
Added checks.